docs(tutorial): add Making a blog with Netlify CMS (#14945)
gatsbyjs/gatsby: docs/tutorial/blog-netlify-cms-tutorial.md

---

> title: How to make a personal Gatsby blog with Netlify CMS

Reviewer: ```suggestion
title: Make a Gatsby Blog with Netlify CMS
```

---

<iframe width="560" height="315" src="https://www.youtube.com/embed/JeTqxCJC56Q" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen title="YouTube video: How to make a fully-featured dev blog in under 10 minutes"></iframe>

Do you want a personal website with a blog running at `your-domain/blog`? Do you like super fast websites with out-of-the-box performance, SEO, and best practices? Do you like automating your work with a straightforward CMS? Well you're in luck!

> This tutorial will use [gatsby-personal-starter-blog](http://t.wang.sh/gatsby-personal-starter-blog), a Gatsby starter that I made that is based on the very popular [gatsby-starter-blog](/starters/gatsbyjs/gatsby-starter-blog/). The differences being that `gatsby-personal-starter-blog` is configured to run the blog on a subdirectory `/blog`, and comes pre-installed with [Netlify CMS](https://www.netlifycms.org/), and VS Code highlighting for code blocks.

Reviewer: ```suggestion
This tutorial will use [gatsby-personal-starter-blog](http://t.wang.sh/gatsby-personal-starter-blog), a Gatsby starter based on the very popular [gatsby-starter-blog](/starters/gatsbyjs/gatsby-starter-blog/). The differences are that `gatsby-personal-starter-blog` is configured to run the blog on a subdirectory `/blog`, and comes pre-installed with [Netlify CMS](https://www.netlifycms.org/), with VS Code highlighting for code blocks.
```

> ## Let's walk through setting up a site in 4 Steps:

Reviewer: ```suggestion
## Walk through setting up a site in 4 steps:
```

### Step 1

Open your Terminal and run the following command from the Gatsby CLI to create a new Gatsby site using [gatsby-personal-starter-blog](http://t.wang.sh/gatsby-personal-starter-blog).

```sh
gatsby new [your-project-name] https://github.com/thomaswangio/gatsby-personal-starter-blog
```

### Step 2

Once the Gatsby site is finished installing all the packages and dependencies, you can now go into the directory and run the site locally.

```sh
cd [your-project-name]
gatsby develop
```

> You'll notice now you can go to `[localhost:8000](http://localhost:8000)` to see your new site, but what's cool is that Netlify CMS is pre-installed, and you can access it at `localhost:8000/admin`. A CMS, or content management, is useful because you can add content like blog posts from a dashboard on your site instead of having to add posts manually. However, we want to be able to access the CMS from a deployed website, not just locally. For that, we'll need to deploy to Netlify, set up continuous deployment, and do a few configurations.

Reviewer: ```suggestion
You'll notice now you can go to `[localhost:8000](http://localhost:8000)` to see your new site, but what's cool is that Netlify CMS is pre-installed, and you can access it at `localhost:8000/admin`. A CMS, or content management system, is useful because you can add content like blog posts from a dashboard on your site instead of having to add posts manually with Markdown. However, you'll likely want to be able to access the CMS from a deployed website, not just locally. For that, you'll need to deploy to Netlify, set up continuous deployment, and do a few configurations.
```

### Step 3

> Open [github.com](http://github.com) and create a new repository, with the same name as your project. Push your new Gatsby site's code to GitHub using the following Terminal commands:

Reviewer: ```suggestion
Create a new [GitHub repository](https://help.github.com/en/articles/create-a-repo) with the same name as your project. Push your new Gatsby site's code to GitHub using the following Terminal commands:
```

```sh
git init
git add .
git commit -m "init"
git remote add origin https://github.com/[your-username]/[your-repo-name].git
git push -u origin master
```

> Then, open [app.netlify.com](http://app.netlify.com) and add a "New site from Git". Choose your newly created repo and click on "Deploy site".

Reviewer: I think this should be step 4! A screenshot of the Netlify interface would help a lot.

Author: done! thanks for the suggestion

### Step 4

Open the project in your code editor and open `static/admin/config.yml`. Replace the following piece of code with these settings, replacing `your-username/your-repo-name` with your GitHub username and repo. Then push this code to GitHub.

> ```yml

Reviewer: I think the diff format would help here. There's an example in [this doc](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/docs-contributions.md#converting-a-document-from-a-stub).
```suggestion
```diff
```

// Replace this
backend:
  name: test-repo

// With this
backend:
  name: github
  repo: your-username/your-repo-name
```

> To make sure that Netlify CMS has access to your GitHub, you need to set up an OAuth application on GitHub. The instructions for that are here: [https://www.netlify.com/docs/authentication-providers/#using-an-authentication-provider](https://www.netlify.com/docs/authentication-providers/#using-an-authentication-provider). Once you've configured an authentication provider then you'll be able to use Netlify CMS at your deployed site to add new posts.

Reviewer: ```suggestion
To make sure that Netlify CMS has access to your GitHub repo, you need to set up an OAuth application on GitHub. The instructions for that are here: [https://www.netlify.com/docs/authentication-providers/#using-an-authentication-provider](https://www.netlify.com/docs/authentication-providers/#using-an-authentication-provider). Once you've configured an authentication provider then you'll be able to use Netlify CMS at your deployed site to add new posts.
```
If you have a screenshot of what this process would look like, that would help!

Author: done! added the example one from the official docs https://github.com/gatsbyjs/gatsby/pull/14945/commits/d315cf4444d2cf623731d110ba345017f7431806

## Customizing your site

Head into `gatsby-config.js` and you can edit your siteMedata, add a Google Analytics tracking ID, and your app icon/favicon.

> To connect your Netlify site to your custom domain, see [Netlify's instructions on custom domains](https://www.netlify.com/docs/custom-domains/). If you want to learn more about Gatsby, you can head over to the [Gatsby documentation](/docs/) and learn how to work with data and their API. To learn more about React to customize the look of your site, the [React documentation](https://reactjs.org/docs/getting-started.html) has many useful references and guides.

Reviewer: ```suggestion
To connect your Netlify site to your custom domain, see [Netlify's instructions on custom domains](https://www.netlify.com/docs/custom-domains/). If you want to learn more about Gatsby, check out the [Gatsby documentation](/docs/) and learn how to work with [sourcing data](/docs/content-and-data/) and [Gatsby APIs](/docs/api-reference/). To learn more about customizing the look of your site, the [Gatsby styling docs](/docs/styling/) and [React documentation](https://reactjs.org/docs/getting-started.html) contain useful references and guides.
```

> Happy hacking and [share with me your new blog](https://twitter.com/intent/tweet?text=%40thomaswangio%20Check%20out%20my%20new%20blog) if you want! Also, you can [subscribe to my Youtube channel](https://t.wang.sh/yt-sub) if you'd like to see more video tutorials.

Reviewer: I think we should remove this from the tutorial to keep it generic enough for the Gatsby tutorials. But as I mentioned above, it would be great to get a blog post from you in the future (and it would be good to mention at the end of a post)! Here is some more info on blog posts: https://www.gatsbyjs.org/contributing/blog-and-website-contributions/
